opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -118,63 +118,118 @@ message MetricDescriptor {
+    // previous measurements like is the case for CUMULATIVE metrics.
+    //
+    // For example, consider a system measuring the number of requests that
+    // it receives and reports the sum of these requests every second as a
+    // DELTA metric:
+    //
+    //   1. The system starts receiving at time=t_0.
+    //   2. A request is received, the system measures 1 request.
+    //   3. A request is received, the system measures 1 request.
+    //   4. A request is received, the system measures 1 request.
+    //   5. The 1 second collection cycle ends. A metric is exported for the
+    //      number of requests received over the interval of time t_0 to
+    //      t_0+1 with a value of 3.
+    //   6. A request is received, the system measures 1 request.
+    //   7. A request is received, the system measures 1 request.
+    //   8. The 1 second collection cycle ends. A metric is exported for the
+    //      number of requests received over the interval of time t_0+1 to
+    //      t_0+2 with a value of 2.
+    DELTA = 2;
+
+    // CUMULATIVE is a metric whose values are the aggregation of
+    // successively made measurements from a fixed start time until the last
+    // reported measurement. This means that current values of a CUMULATIVE
+    // metric depend on all previous measurements since the start time.
+    // Because of this, the sender is required to retain this state in some
+    // form. If this state is lost or invalidated, the CUMULATIVE metric
+    // values MUST be reset and a new fixed start time following the last
+    // reported measurement time sent MUST be used.
+    //
+    // For example, consider a system measuring the number of requests that
+    // it receives and reports the sum of these requests every second as a
+    // CUMULATIVE metric:
+    //
+    //   1. The system starts receiving at time=t_0.
+    //   2. A request is received, the system measures 1 request.
+    //   3. A request is received, the system measures 1 request.
+    //   4. A request is received, the system measures 1 request.
+    //   5. The 1 second collection cycle ends. A metric is exported for the
+    //      number of requests received over the interval of time t_0 to
+    //      t_0+1 with a value of 3.
+    //   6. A request is received, the system measures 1 request.
+    //   7. A request is received, the system measures 1 request.
+    //   8. The 1 second collection cycle ends. A metric is exported for the
+    //      number of requests received over the interval of time t_0 to
+    //      t_0+2 with a value of 5.
+    //   9. The system experiences a fault and loses state.
+    //   10. The system recovers and resumes receiving at time=t_1.
+    //   11. A request is received, the system measures 1 request.
+    //   12. The 1 second collection cycle ends. A metric is exported for the
+    //      number of requests received over the interval of time t_1 to
+    //      t_0+1 with a value of 1.
+    CUMULATIVE = 3;
```

Comment:
One edge case that probably we should document is if the implementation decides to "reset" the `start_time` every export interval making this type essentially a `delta`. What should we recommend/do when we see that case?

Reply:
This would be the case for Stateless Batcher from Metric Specs.

Reply:
I think the Stateless Batcher will mark that as a proper `Delta`.

Reply:
> One edge case that probably we should document is if the implementation decides to "reset" the `start_time` every export interval making this type essentially a `delta`. What should we recommend/do when we see that case?

I'm not sure how to, nor if we want to, address all the possible ways this can be misused. I think in the situation you describe it would absolutely be an example of incorrectly defining the temporality of the data being sent, and it would not match the documentation already given here, but I don't know how we could handle this other than just processing/resetting every cycle.

Reply:
Just probably for the moment document clearly that this is a misused of the property.

Reply:
It sounds like misuse, and it probably will cause poor performance in a downstream system, but the downstream system should be able to handle it correctly according to these definitions. A new start time implies a new cumulative series, so we can see this case as a progression of single-entry series representative deltas of a cumulative instrument.

Reply:
@MrAlias please maybe just file an issue where you save the current comments, and we can followup. I would prefer not to forget about this discussion.

Reply:
@bogdandrutu : created https://github.com/open-telemetry/opentelemetry-proto/issues/143

I'll be sure to address later.